pxr/usd/lib/sdf/assetPathResolver.cpp

```diff
@@ -209,9 +209,15 @@ string
 Sdf_GetAnonLayerDisplayName(
     const string& identifier)
 {
-    if (std::count(identifier.begin(), identifier.end(), ':') == 2)
-        return identifier.substr(identifier.rfind(':') + 1);
-    return std::string();
+    // We want to find the second occurence of ':', traversing from the left,
+    // in our identifier which is of the form anon:0x4rfs23:displayName
+    auto fst = std::find(identifier.begin(), identifier.end(), ':');
```

Comment:
Can we bail here if fst == identifier.end() for safety? The old code handled the case where there were no ":" characters in the identifier (even if this should never happen), so I think it'd be good to follow suit.

Reply:
Done, updated. Thanks :)